spring-integration-core/src/main/java/org/springframework/integration/config/xml/DefaultInboundChannelAdapterParser.java

```diff
@@ -121,7 +119,7 @@ else if (hasRef) {
 				result = sourceValue;
 			}
 		}
-		return result;
+		return Objects.requireNonNull(result);
```

Comment:
Same here: why not exception after `error()`?

Reply:
There is a path where result is never set.   Unless we make the assumption that the check on line 118 can be removed.